src/diffpy/labpdfproc/tests/test_tools.py

```diff
@@ -76,3 +77,44 @@ def test_set_wavelength_bad(inputs, msg):
     actual_args = argparse.Namespace(wavelength=inputs[0], anode_type=inputs[1])
     with pytest.raises(ValueError, match=re.escape(msg[0])):
         actual_args.wavelength = set_wavelength(actual_args)
+
+
+params5 = [
+    ([None], []),
+    (
+        [["facility=NSLS II", "beamline=28ID-2", "favorite color=blue"]],
+        [["facility", "NSLS II"], ["beamline", "28ID-2"], ["favorite color", "blue"]],
+    ),
+    ([["facility=NSLS II", "facility=NSLS III"]], [["facility", "NSLS III"]]),
```

Comment:
This is a great test (I like how you are thinking here), but might it not be better to raise an error in this case?  Otherwise, like the git example we discussed before, you are having the code make decisions for the human and that breaks our magic rule.  Better to let the person know there is a boo boo and ask them to rerun fixing the boo boo (don't use the words "boo boo"!)

Reply:
Oh this makes sense! I updated and made it an error.